src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int capacity = 10;
+    private int size;
+
+    private T[] elementData;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[capacity];
+    }
+
+    public ArrayList(int arrayListCapacity) {
+        this.elementData = (T[]) new Object[arrayListCapacity];
+        capacity = arrayListCapacity;
+    }
+
+    private void increaseCapacity() {
+        if (this.size == elementData.length) {
+            capacity = (capacity * 3) / 2 + 1;
+            T[] tempElementData = (T[]) new Object[capacity];
+            System.arraycopy(elementData, 0, tempElementData, 0, size);
+            elementData = tempElementData;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        increaseCapacity();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            increaseCapacity();
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+            elementData[index] = value;
+            size++;
+            return;
+        }
+        throw new ArrayListIndexOutOfBoundsException(index + " is invalid index");
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list.size() > this.capacity - this.size) {
+            capacity = this.capacity - this.size + list.size();
+            T[] tempElementData = (T[]) new Object[capacity];
+            System.arraycopy(elementData, 0, tempElementData, 0, this.size);
+            elementData = tempElementData;
+            for (int i = 0; i < list.size(); i++) {
+                elementData[this.size + i] = list.get(i);
+            }
+            this.size += list.size();
+        } else {
+            for (int i = 0; i < list.size(); i++) {
+                elementData[this.size + i] = list.get(i);
+            }
+            this.size += list.size();
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        for (int i = 0; i < size; i++) {
```

Comment:
get() method should have constant algorithm complexity. check the index - if it's in bounds just `return elementData[index]` without iteration